[storage-file-share] README files for samples directories (#6208)
Azure/azure-sdk-for-js: sdk/storage/storage-file-share/samples/typescript/README.md

---
page_type: sample
languages:
- typescript
products:
- azure
- azure-storage
urlFragment: storage-file-share-typescript
---

# Azure Storage File Share client library samples for TypeScript

These sample programs show how to use the TypeScript client libraries for Azure Storage File Shares in some common scenarios.

|__File Name__|__Description__|
|-------------|---------------|

> |[basic.ts][basic]|authenticate with the service using an account name & key (or anonymously with a SAS URL), upload a file, list files and directories, and download a file to a string|

Reviewer: same English nit as JS

|[withConnString.ts][withConnString]|connect to and authenticate with the service using a connection string|
|[sharedKeyCred.ts][sharedKeyCred]|authenticate with the service using an account name and a shared key|
|[anonymousCred.ts][anonymousCred]|authenticate with the service anonymously using a SAS URL|
|[proxyAuth.ts][proxyAuth]|connect to the service using a proxy and authenticate with an account name & key|
|[iterators-shares.ts][iterators-shares]|connect to the service and iterate through the shares in the account|
|[iterators-files-and-directories.ts][iterators-files-and-directories]|create a few directories and iterate through them individually (using async `for await` syntax), by page, and resuming paging using a marker|
|[iterators-handles.ts][iterators-handles]|connect to the service and iterate through open handles|
|[customPipeline.ts][customPipeline]|use custom HTTP pipeline options when connecting to the service|
|[advanced.ts][advanced]|use custom logging and pipeline options, then upload a local file to a share|

## Prerequisites

> The sample code is compatible with Node.js. The samples are compatible with Node.js >= 8.0.0, except the samples that use the async `for await` syntax. Samples that use the async `for await` syntax require a version of Node that is compatible with that syntax (>= 10.0.0).

Reviewer: ```suggestion
Samples are compatible with Node.js >= 8.0.0, except for samples which use `for await` syntax, which require Node >= 10.0.0.
```

Before running the samples in Node, they must be compiled to JavaScript using the TypeScript compiler. For more information on TypeScript, see the [TypeScript documentation][typescript]. Install the TypeScript compiler using

```bash
npm install -g typescript
```

You need [an Azure subscription][freesub] and [an Azure Storage account][azstorage] to run these sample programs. Samples retrieve credentials to access the storage account from environment variables. See each individual sample for details on which environment variables it requires to function.

Adapting the samples to run in the browser requires some additional consideration. For details, please see the [package README][package].

## Setup

To run the samples using the published version of the package:

1. Install the client library using `npm`:
```bash
npm install @azure/storage-file-share
```

> 2. Copy and modify the sample code, replacing the relative `import { ... } from '../../src'` statements with `import { ... } from '@azure/storage-file-share'`

Reviewer: ```suggestion
2. Copy and modify the sample code, replacing the relative `import { ... } from "../../src"` statements with `import { ... } from "@azure/storage-file-share"`
```
